Add a quirk for gmail on ApplicationCache

szewai opened the pull request:
#### ba526d288906723d792880e7b1a803f19cf80006
<pre>
Add a quirk for gmail on ApplicationCache
<a href="https://bugs.webkit.org/show_bug.cgi?id=243298">https://bugs.webkit.org/show_bug.cgi?id=243298</a>
rdar://97351877

Reviewed by Chris Dumez.

Enable window.applicationcache on gmail until they fix rdar://97727706.

* Source/WebCore/loader/appcache/DOMApplicationCache.idl:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldEnableApplicationCacheQuirk const):
* Source/WebCore/page/Quirks.h:

Canonical link: <a href="https://commits.webkit.org/252942@main">https://commits.webkit.org/252942@main</a>
</pre>

cdumez commented on Source/WebCore/page/Quirks.h:
```diff
@@ -211,6 +212,7 @@ class Quirks {
     mutable std::optional<bool> m_allowLayeredFullscreenVideos;
 #endif
     mutable std::optional<bool> m_hasBrokenPermissionsAPISupportQuirk;
+    mutable std::optional<bool> m_shouldEnableApplicationCacheQuirk;
```
Can be guarded by `#if PLATFORM(IOS_FAMILY)` I believe.

szewai replied:
Updated; thanks for the review

szewai committed:
Add a quirk for gmail on ApplicationCache
https://bugs.webkit.org/show_bug.cgi?id=243298
rdar://97351877

Reviewed by Chris Dumez.

Enable window.applicationcache on gmail until they fix rdar://97727706.

* Source/WebCore/loader/appcache/DOMApplicationCache.idl:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldEnableApplicationCacheQuirk const):
* Source/WebCore/page/Quirks.h:

Canonical link: https://commits.webkit.org/252942@main
